operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -870,6 +874,40 @@ func (r *RedpandaReconciler) reconcileDelete(ctx context.Context, rp *redpandav1
 	return nil
 }
 
+func (r *RedpandaReconciler) deleteHelmChart(ctx context.Context, rp *redpandav1alpha2.Redpanda) error {
+	var hr helmv2beta2.HelmRelease
+	hrName := rp.GetHelmReleaseName()
+
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: rp.Namespace, Name: hrName}, &hr); err != nil {
+		// Ignore any error related to retrieving HelmRelease
+		return nil
+	}
+
+	// Try to delete HelmChart custom resource only when HelmRelease is suspended and HelmChart from status is not empty
+	// Reference:
+	// https://github.com/fluxcd/helm-controller/blob/2d335f2aa0e2e0df2a631ebf19394aed07c556f3/internal/reconcile/helmchart_template.go#L163-L166
+	// https://github.com/fluxcd/helm-controller/blob/2d335f2aa0e2e0df2a631ebf19394aed07c556f3/internal/controller/helmrelease_controller.go#L375-L388
+	if hr.Spec.Suspend && hr.Status.HelmChart != "" {
+		namespacedName := client.ObjectKey{Namespace: rp.Namespace, Name: hr.Status.HelmChart}
+		var chart sourcev1.HelmChart
+		err := r.Client.Get(ctx, namespacedName, &chart)
+		if err != nil && !apierrors.IsNotFound(err) {
+			// Return error to retry until we succeed.
+			err = fmt.Errorf("failed to delete HelmChart '%s': %w", namespacedName.Name, err)
```

Comment:
This error isn't related to deletion, please update the error message

Reply:
thx fixed